juju/main.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import yaml
+
+# Use modules from the $JUJU_CHARM_DIR/lib directory. An absolute path
+# is used as using a symlink like hooks/install -> ../lib/juju/main.py
+# will result in $JUJU_CHARM_DIR/lib/juju being the first path stored
+# in sys.path during hook execution which is undesirable.
+# This happens because the directory containing a script is added to
+# sys.path by the interpreter which gets resolved if it is a symlink.
+del sys.path[0]
+sys.path.insert(0, os.path.join(os.environ.get("JUJU_CHARM_DIR", ''), 'lib'))
+
+import juju.framework # noqa
+
+from juju.framework import ( # noqa
+    Framework,
+    Event,
+    BoundEvent
+)
+from charm import Charm # noqa
+
+# Reuse the same db file as used by charm-helpers.
+DB_PATH = os.path.join(
+    os.environ.get('CHARM_DIR', ''), '.unit-state.db')
+
+try:
+    from charmhelpers.core.hookenv import log
+except (ImportError, ModuleNotFoundError):
+    log = print
+
+
+def load_metadata():
```

Comment:
Let's please underline all of these methods other than main() for now. We don't want people thinking this is part of the API yet, and most of these will need to be twisted into something more reasonable in due time (any function that looks at $PWD is suspect).